### PR TITLE
Controller changes to get correct id token based on V1/V2

### PR DESCRIPTION
**Pull request by kreedula**
- Changes tested with both Msal and Adal test apps, works as expected
- Minor code improvements

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java [hunk 1]
```diff
@@ -222,8 +222,8 @@ public ICacheRecord save(@NonNull AccountRecord accountRecord,
         }
 
         updateApplicationMetadataCache(
-                result.getIdToken().getClientId(),
-                result.getIdToken().getEnvironment(),
+                result.getAccessToken().getClientId(),
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java]: 💪

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OpenIdConnectPromptParameter.java [hunk 2]
```diff
@@ -64,12 +62,12 @@ public String toString() {
 
     /**
      * Utility method to map Adal PromptBehavior with OpenIdConnectPromptParameter
-     * @param promptbehavior
+     * @param promptBehavior
      * @return
      */
-    public static OpenIdConnectPromptParameter _fromPromptBehavior(@NonNull final String promptbehavior){
+    public static OpenIdConnectPromptParameter _fromPromptBehavior(final String promptBehavior){
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OpenIdConnectPromptParameter.java]: If you were so inclined we could now annotate this as `@Nullable`

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 3]
```diff
@@ -70,7 +71,7 @@ public BrokerRequest brokerRequestFromSilentOperationParameters(@NonNull final A
                 .localAccountId(parameters.getAccount().getLocalAccountId())
                 .username(parameters.getAccount().getUsername())
                 .claims(parameters.getClaimsRequestJson())
-                .forceRefresh(parameters.getForceRefresh())
+                .forceRefresh(parameters.getForceRefresh() || !TextUtils.isEmpty(parameters.getClaimsRequestJson()))
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: These changes are related or fixing another issue?

Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: yes , have a vsts issue, will link the PR there

**Pull request merged**